src/analog/private/m2kanalogout_impl.cpp

```diff
@@ -312,12 +349,29 @@ class M2kAnalogOut::M2kAnalogOutImpl : public DeviceGeneric {
                         throw_exception(EXC_INVALID_PARAMETER, "Analog Out: Input array length must be multiple of channels");
                 }
 		std::vector<std::vector<short>> data_buffers;
-		m_m2k_fabric->setBoolValue(0, false, "powerdown", true);
-		m_m2k_fabric->setBoolValue(1, false, "powerdown", true);
-		setSyncedDma(true);
+		unsigned int bufferSize = nb_samples/nb_channels;
+		bool streamingData = true;
+		bool isBufferEmpty = true;
 
-		m_samplerate.at(0) = getSampleRate(0);
-		m_samplerate.at(1) = getSampleRate(1);
+		for (unsigned int  chn = 0; chn < nb_channels; chn++) {
+			streamingData &= !getCyclic(chn);
+		}
+		if (streamingData) {
+			// all kernel buffers are empty when maximum buffer space is equal with the unused space
+			unsigned int unusedBufferSpace, maxBufferSpace;
+			for (unsigned int  chn = 0; chn < nb_channels; chn++) {
+				m_dac_devices.at(chn)->initializeBuffer(bufferSize, false);
+				unusedBufferSpace = m_dac_devices[chn]->getBufferLongValue("data_available");
+				maxBufferSpace = 2u * bufferSize * (m_nb_kernel_buffers.at(chn) - 1);
+				isBufferEmpty &= (maxBufferSpace == unusedBufferSpace);
+			}
+			if (isBufferEmpty) {
+				// when kernel buffers are empty both channels must be synchronized
+				setSyncedDma(true);
+			}
+		} else {
+			setSyncedDma(true);
+		}
 
 		for (unsigned int chn = 0; chn < nb_channels; chn++) {
 			std::vector<short> raw_data_buffer = {};
```

Comment:
Can we use bufferSize on the next line instead of nb_samples/nb_channels?

Reply:
Modified in commit 85c9a60.